build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -439,6 +439,115 @@ service ContentAddressableStorage {
   rpc GetTree(GetTreeRequest) returns (stream GetTreeResponse) {
     option (google.api.http) = { get: "/v2/{instance_name=**}/blobs/{root_digest.hash}/{root_digest.size_bytes}:getTree" };
   }
+
+  // Split a blob into chunks.
+  //
+  // This call splits a blob into chunks, stores the chunks in the CAS, and
+  // returns a list of the chunk digests. Using this list, a client can check
+  // which chunks are locally available and just fetch the missing ones. The
+  // desired blob can be assembled by concatenating the fetched chunks in the
+  // order of the digests from the list.
+  //
+  // This rpc can be used to reduce the required data to download a large blob
+  // from CAS if chunks from earlier downloads of a different version of this
+  // blob are locally available. For this procedure to work properly, blobs
+  // SHOULD be split in a content-defined way, rather than with fixed-sized
+  // chunking.
+  //
+  // If a split request is answered successfully, a client can expect the
+  // following guarantees from the server:
+  //  1. The blob chunks are stored in CAS.
+  //  2. Concatenating the blob chunks in the order of the digest list returned
+  //     by the server results in the original blob.
+  //
+  // Servers MAY implement this functionality, but MUST declare whether they
+  // support it or not by setting the
+  // [CacheCapabilities.blob_split_support][build.bazel.remote.execution.v2.CacheCapabilities.blob_split_support]
+  // field accordingly.
+  //
+  // Clients MAY use this functionality, it is just an optimization to reduce
+  // download network traffic, when downloading large blobs from the CAS.
+  // However, clients MUST first check the server capabilities, whether blob
+  // splitting is supported by the server.
+  //
+  // Clients SHOULD verify whether the digest of the blob assembled by the
+  // fetched chunks results in the requested blob digest.
+  //
+  // Since the generated chunks are stored as blobs, they underlie the same
+  // lifetimes as other blobs. In particular, the chunk lifetimes are
+  // independent from the lifetime of the original blob:
+  //  * A blob and any chunk derived from it may be evicted from the CAS at
+  //    different times.
+  //  * A call to Split extends the lifetime of the original blob, and sets
+  //    the lifetimes of the resulting chunks (or extends the lifetimes of
+  //    already-existing chunks).
+  //  * Touching a chunk extends its lifetime, but does not extend the
+  //    lifetime of the original blob.
+  //  * Touching the original blob extends its lifetime, but does not extend
```

Comment:
Similarly to the point above, how about "but the server MAY choose not to extend" ?